### DISCUSSION TITLE
How to help fixing bugs

### DISCUSSION THREAD

**Discussion by florianwachs** [dotnet/maui#10308] [Q&A] [answered]
I love .net and want MAUI to become the best framework. i read the guide on how to build it local. But I still have no real clue how to best tackle a bug, like how do you profile it? it would be nice, at least for me, to have a „how to fix bugs in maui“. I know this is a bit vague, but any ressources that help me to understand how to best tackle issues would be great.

**Comment by pictos** [comment, marked as answer]
Hey @florianwachs, first of all I'm not part of the team, but I've been contributing to the framework since Forms. So, the most difficult part is to get your environment ready to code and debug. After that:
##  1. look for open issues 

First look for issues with the `help-wanted` label. If you see one that you think you can fix, post a comment on the issue letting the team know that you want to work on that. This is important to avoid duplicate work.
If you can't find any good issue with that label, but find another issue that you feel comfortable working on, ask the team and wait for them to say if you can work or not on the issue... Normally they will allow you to work (prioritize bug fixes).

## 2. Open a draft-PR

When you've minimal work you can open a draft PR, that way the team and community can provide feedback early on and help you with any questions about the implementation that you may have.

## 3. Make your PR ready for review

Here your PR will be reviewed by the team and community, address the requested feedback and that will be merged soon. Sometimes take longer than we want, but it's normal in OSS projects. 


If you don't feel comfortable contributing with .NET MAUI directly, but still want to contribute with OSS and .NET MAUI, look at our [CommunityToolkit.MAUI](https://github.com/CommunityToolkit/Maui) and [CommunityToolkit.MAUI.Markup](https://github.com/CommunityToolkit/Maui.Markup) projects. They are simpler to build and get started in the OSS world.

Let me know if you (and anyone else) have more questions (:

**Reply by florianwachs** [reply]
Thank you very much! The community toolkit sounds like an awesome way to get started.

**Reply by florianwachs** [reply]
@pictos one follow up question, do you have guidance on how to do performance analysis with Maui too? For example I recently tried and awesome looking Maui sample but the list was very janky on an Samsung S21 ultra. Would be awesome to know how to tackle such a problem. Have a great day

**Reply by pictos** [reply]
@florianwachs I know about these two resources:
- https://github.com/dotnet/maui/wiki/Profiling-.NET-MAUI-Apps
- https://github.com/jonathanpeppers/maui-profiling

**Reply by florianwachs** [reply]
Awesome, thank you very much
